storefront/test/helpers/workarea/api/storefront/application_helper_test.rb

```diff
@@ -0,0 +1,18 @@
+require 'test_helper'
+
+module Workarea
+  module Api
+    module Storefront
+      class ApplicationHelperTest < ViewTest
+        def test_storefront_api_url_for
+          navigable = create_page
+          taxon = create_taxon(navigable: navigable)
+
+          expects(:page_url).with(navigable.slug, port: nil)
```

Comment:
This is the easiest way I could figure out to test this. I tried including `Rails.application.routes.url_helpers` and `Workarea::Storefront::Engine.routes.url_helpers`, but wasn't able to get past the "Missing host to link to" error you get when `#default_url_options` aren't set. This is strange, because I do see them being set in the `Workarea::ViewTest` base class. I wasn't really sure why this is happening, and this test isn't super future-proof because if for some reason Rails changes their mind about `port: nil` being acceptable in the future, this test will still pass and the problem will reveal itself again (possibly) once someone deploys the API to QA.

Generally, I try to stay away from mocking the value out in tests, but in this case I feel like it's an acceptable risk. However, I will definitely go deeper and try to actually test the output of the route if you guys feel like it's necessary for this project's ongoing health.

Reply:
You could define a `default_url_options` method in this class.

Reply:
that legit did not work a few days ago and now it does and i am confused :D